app/controllers/TaxAccountSummaryController.scala

```diff
@@ -76,19 +76,31 @@ class TaxAccountSummaryController @Inject()(trackingService: TrackingService,
           }
   }
 
-  private def taxAccountSummaryViewModel(nino: Nino, taxAccountSummary: TaxAccountSummary)(implicit hc: HeaderCarrier, messages: Messages) = {
+  private def taxAccountSummaryViewModel(nino: Nino, taxAccountSummary: TaxAccountSummary)
+                                        (implicit hc: HeaderCarrier, messages: Messages): Future[TaxAccountSummaryViewModel] =
     for {
-      taxCodeIncomes <- taxAccountService.taxCodeIncomes(nino, TaxYear())
+      livePensionIncomeSources <- taxAccountService.incomeSources(nino, TaxYear(), "PensionIncome", "Live")
+      liveEmploymentIncomeSources <- taxAccountService.incomeSources(nino, TaxYear(), "EmploymentIncome", "Live")
+      ceasedEmploymentIncomeSources <- taxAccountService.incomeSources(nino, TaxYear(), "EmploymentIncome", "Ceased")
+      nonMatchingCeasedEmployments <- employmentService.ceasedEmployments(nino, TaxYear())
       nonTaxCodeIncome <- taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
-      employments <- employmentService.employments(nino, TaxYear())
       isAnyFormInProgress <- trackingService.isAnyIFormInProgress(nino.nino)
     } yield {
-      (taxCodeIncomes, nonTaxCodeIncome) match {
-        case (TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]),
+      (livePensionIncomeSources, liveEmploymentIncomeSources, ceasedEmploymentIncomeSources, nonMatchingCeasedEmployments, nonTaxCodeIncome) match {
+        case (TaiSuccessResponseWithPayload(livePensionIncomeSources: Seq[IncomeSource]),
+        TaiSuccessResponseWithPayload(liveEmploymentIncomeSources: Seq[IncomeSource]),
+        TaiSuccessResponseWithPayload(ceasedEmploymentIncomeSources: Seq[IncomeSource]),
+        nonMatchingCeasedEmployments: Seq[Employment],
         TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome)) =>
-          TaxAccountSummaryViewModel(taxCodeIncomes, employments, taxAccountSummary, isAnyFormInProgress, nonTaxCodeIncome)(messages)
+          TaxAccountSummaryViewModel(taxAccountSummary,
+            isAnyFormInProgress,
+            nonTaxCodeIncome,
+            livePensionIncomeSources,
+            liveEmploymentIncomeSources,
+            ceasedEmploymentIncomeSources,
+            nonMatchingCeasedEmployments)(messages)
         case _ => throw new RuntimeException("Failed to fetch income details")
```

Comment:
Do you need to pass all these parameters? Maybe merging them and pushing them into an IncomeSources object would tidy this up. 

Don't need the implicit (messages) at the end here

Reply:
Removed the implicit message. I personally prefer it like this as it's clear what each value is as it creates the model

Reply:
however, I will merge into IncomeSources